datafusion/sqllogictest/test_files/insert_to_external.slt

```diff
@@ -76,6 +76,44 @@ select * from dictionary_encoded_parquet_partitioned order by (a);
 a foo
 b bar
 
+statement ok
+CREATE EXTERNAL TABLE dictionary_encoded_arrow_partitioned(
+  a varchar,
+  b varchar,
+) 
+STORED AS arrow
+LOCATION 'test_files/scratch/insert_to_external/arrow_dict_partitioned/'
+PARTITIONED BY (b)
+OPTIONS(
+create_local_path 'true',
+insert_mode 'append_new_files',
+);
+
+query TT
+insert into dictionary_encoded_arrow_partitioned 
+select * from dictionary_encoded_values
+----
+2
+
+statement ok
+CREATE EXTERNAL TABLE dictionary_encoded_arrow_test_readback(
+  a varchar,
+) 
+STORED AS arrow
+LOCATION 'test_files/scratch/insert_to_external/arrow_dict_partitioned/b=bar/'
+OPTIONS(
+create_local_path 'true',
+insert_mode 'append_new_files',
+);
+
+query T
+select * from dictionary_encoded_arrow_test_readback;
+----
+b
+
+query error DataFusion error: Arrow error: Schema error: project index 1 out of bounds, max field 1
```

Comment:
Reading a table of partitioned arrow files fails with this error. I believe the arrow files themselves are fine as shown in the test above that reads one of the partitions correctly.

Reply:
Possibly similar cause as #7816 ?

Reply:
```suggestion
# https://github.com/apache/arrow-datafusion/issues/7816
query error DataFusion error: Arrow error: Schema error: project index 1 out of bounds, max field 1
```